R/layer-sf.R

```diff
@@ -62,3 +74,9 @@ is_sf <- function(data) {
 #' @export
 scale_type.sfc <- function(x) "identity"
 
+# helper function to determine the geometry type of sf object
+sf_geometry_type <- function(sf) {
+  geometry_type <- unique(as.vector(sf::st_geometry_type(sf)))
+  if (length(geometry_type) != 1)  geometry_type <- "blend"
```

Comment:
I still don't like `"blend"`. It is introduced here without documentation and not used anywhere else. I'd be fine with `"other"`, since that is used elsewhere in the code.